### ISSUE TITLE
Changing beam properties on the first note of a system can cause note positioning issues

### ISSUE THREAD

**Issue by Panogen** [musescore/MuseScore#21116]
### Issue type

Engraving bug (incorrect score rendering)

### Bug description

Setting either "Break inner beams (eighth)", "Break inner beams (16th)", or "Join beam" on the first note of a system causes the following chord to render incorrectly, but only if the last note on the previous system is not set to "No beam". This also occurs if the last rest of the previous system is set to anything other than "No beam".

### Steps to reproduce

Enter an 8th note at the end of a measure, then insert a line break at the end of that measure. On the new system, enter two consecutive eighth notes starting at the first beat of the measure. The second should be a chord with any kind of 2nd in it (such as A4 and B4). Drag the "Join beam" beam property from the palette onto the first note on the new system.

### Screenshots/Screen recordings

Bug demonstration

https://github.com/musescore/MuseScore/assets/73259774/1a226f00-915c-4b3e-93e6-5e8001e4b300

### MuseScore Version

MuseScore version (64-bit): 4.2.0-233521124, revision: github-musescore-musescore-eb8d33c

### Regression

I don't know

### Operating system

Windows 11

**Comment by oktophonie** [comment]
I can't reproduce this problem in a minimal example following your reproduction steps, so there may be some context missing. Perhaps you might attach the file from the video (which clearly demonstrates the issue)

**Comment by Panogen** [comment]
Surprisingly, after restarting my computer, it seems that the issue has gone away. I am no longer able to reproduce it on the measures shown in the video or on a brand-new file. I did make some edits before trying to make the error happen again, though, so that might be what caused the issue to go away. Regardless, here is the current version of the file (I no longer have the version shown in the video): 
[1-9-24.zip](https://github.com/musescore/MuseScore/files/14002785/1-9-24.zip)

It might still be worth investigating since yesterday the problem persisted even after restarting the app.

**Comment by oktophonie** [comment]
Yes, I can reproduce it in the file by fiddling on with the beam options:
![image](https://github.com/musescore/MuseScore/assets/20604032/caa0fb09-d6b2-4c4a-948e-8669b36123e2)
will investigate.
